hashes/src/sha512.rs

```diff
@@ -79,22 +79,30 @@ impl crate::HashEngine for HashEngine {
 
 /// Output of the SHA512 hash function.
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
 #[repr(transparent)]
-pub struct Hash(
-    #[cfg_attr(
-        feature = "schemars",
-        schemars(schema_with = "crate::util::json_hex_string::len_64")
-    )]
-    [u8; 64],
-);
+pub struct Hash([u8; 64]);
 
 impl Hash {
     fn internal_new(arr: [u8; 64]) -> Self { Hash(arr) }
 
     fn internal_engine() -> HashEngine { Default::default() }
 }
 
+#[cfg(feature = "schemars")]
+impl schemars::JsonSchema for Hash {
+    fn schema_name() -> String { "Hash".to_owned() }
+
+    fn json_schema(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        let mut schema: schemars::schema::SchemaObject = <String>::json_schema(gen).into();
```

Comment:
`sha512::Hash::json_schema(gen)`

Reply:
This one is needed as is, this _is_ the implementation of `json_schema` for `512::Hash`. You should be proud of yourself though, I tried to build it anyways just in case there was some Kix magic that I was overlooking :)

Reply:
Why isn't this one produced by the macro though?

Reply:
Ha! We are not using the macro, and for no good reason. I'll fix that then rebase this on top of it. Good eye!

Reply:
I suspect it had something to do with large arrays not implementing some traits but that should now no longer be an issue.